### PR TITLE
Refactor urls and add missing app tests for location

### PR DESCRIPTION
**Pull request by web-bert**
Remove snake case for location methods and refactor urls to group them
Add missing tests for setting location on a barrier
Add missing tests for setting location on a report
Fix bug where save and exit did not work on report for saving location
Update fake metadata to make admin areas work in tests

### CODE REVIEW

**Comment by web-bert** [comment]
@CWaszczuk-DIT bit of a bigger refactor than I had imagined

File: src/app/sub-apps/barriers/routes.js [hunk 1]
```diff
@@ -57,15 +57,14 @@ module.exports = ( express, app ) => {
...
-	app.get( '/:barrierId/location/admin-areas/add', controller.location.add_admin_area );
-	app.post( '/:barrierId/location/admin-areas/add', controller.location.add_admin_area );
-	app.get( '/:barrierId/location/admin-areas/remove', controller.location.remove_admin_area );
-	app.post( '/:barrierId/location/admin-areas/remove', controller.location.remove_admin_area );
-	
+	app.get( '/:barrierId/location/edit/', controller.location.edit );
+	app.post( '/:barrierId/location/edit/', controller.location.edit );
+	app.get( '/:barrierId/location/country/', controller.location.country );
+	app.post( '/:barrierId/location/country/', controller.location.country );
+	app.get( '/:barrierId/location/admin-areas/', controller.location.adminAreas.add );
```
Comment (CWaszczuk-DIT) [inline src/app/sub-apps/barriers/routes.js]: Despite keeping urls without breaking parts if you remove, I don't feel like it's a suitable pattern when it doesn't describe the page you're looking at. My initial though is `/:barrierId/location/admin-areas/` would be a list page of all the admin areas on the barrier, but it links to an add page. Think this should be /add for the sake of consistency with remove and the purpose of the page.

Comment (web-bert) [reply src/app/sub-apps/barriers/routes.js]: Technically it does show a list of admin areas. I guess the difference here is that on the barrier detail the country and admin areas are shown together, but in the add flow they are separate pages.

**Comment by web-bert** [comment]
@CWaszczuk-DIT updated the URLs as discussed

**Pull request merged**